assets/src/edit-story/components/inspector/document/publish/index.js

```diff
@@ -99,6 +113,31 @@ function PublishPanel() {
     [updateStory]
   );
 
+  const generateCanvas = useCallback(async () => {
+    try {
+      const canvas = await html2canvas(fullbleedContainer);
+      const blob = await new Promise((resolve, reject) =>
+        canvas.toBlob(
+          (value) => (value ? resolve(value) : reject(new Error())),
+          'image/jpeg'
+        )
+      );
+      const { id, media_details, source_url } = await uploadFile(
+        new File([blob], 'cover-image.jpg', { type: 'image/jpeg' })
```

Comment:
Can we include the story ID in the file name? and then maybe do something like `web-story-1234-cover-generated.jpg`

Makes it clear where the image is coming from (Web Stories), which story it belongs too, and that it's an auto-generated image.

Reply:
Good idea!